python/orca/src/bigdl/orca/learn/tf2/spark_runner.py

```diff
@@ -260,6 +262,15 @@ def __init__(self,
                                 self.model = self.model_creator(self.config)
                                 if self.model_weights:
                                     self.model.set_weights(self.model_weights.value)
+                                if self.optimizer_weights:
+                                    def load_opt_weights():
+                                        grad_vars = self.model.trainable_weights
+                                        zero_grads = [tf.zeros_like(w) for w in grad_vars]
+                                        self.model.optimizer.apply_gradients(
+                                            zip(zero_grads, grad_vars))
+                                        self.model.optimizer.set_weights(
+                                            self.optimizer_weights.value)
+                                    self.strategy.run(load_opt_weights)
```

Comment:
what's the difference between calling strategy.run and directly calling this like setting model weights?

Reply:
it's not allowed to call `model.optimizer.set_weights` directly inside the strategy scope.